### ISSUE TITLE
[R Package] Couldn't get First_Metric_Only functionality

### ISSUE THREAD

**Issue by TpTheGreat** [microsoft/LightGBM#2765]
Hi,
I couldn't get the First_Metric_Only functionality working.
also didn't find any callback for early_stopping to implement it in.

Right now when putting both Validation and Training data in 'Valids' (i use training there to get the rounds loss information), the early stopping stops when Validation stops improving, but best_iter is not updated accordingly

**Comment by guolinke** [comment]
I think it is only supported in python/cli. ping @jameslamb to confirm.

**Comment by StrikerRUS** [comment]
That's true. Please refer to #2368.
